lib/matplotlib/figure.py

```diff
@@ -3176,24 +3236,24 @@ def tight_layout(self, *, pad=1.08, h_pad=None, w_pad=None, rect=None):
 
         See Also
         --------
-        .Figure.set_tight_layout
+        .Figure.set_layout_engine
         .pyplot.tight_layout
         """
-        from contextlib import nullcontext
-        from .tight_layout import (
-            get_subplotspec_list, get_tight_layout_figure)
+        from .tight_layout import get_subplotspec_list
         subplotspec_list = get_subplotspec_list(self.axes)
         if None in subplotspec_list:
             _api.warn_external("This figure includes Axes that are not "
                                "compatible with tight_layout, so results "
                                "might be incorrect.")
-        renderer = _get_renderer(self)
-        with getattr(renderer, "_draw_disabled", nullcontext)():
-            kwargs = get_tight_layout_figure(
-                self, self.axes, subplotspec_list, renderer,
-                pad=pad, h_pad=h_pad, w_pad=w_pad, rect=rect)
-        if kwargs:
-            self.subplots_adjust(**kwargs)
+        # note that here we do not _set_ the figures engine to tight_layout
+        # but rather just perform the layout in place for back compatibility.
+        engine = TightLayoutEngine(pad=pad, h_pad=h_pad, w_pad=w_pad,
+                                   rect=rect)
+        try:
+            self.set_layout_engine(engine)
```

Comment:
The note above says we do not have to do this?  Is we are going to swap like this, should we be caching what it was and setting it back or is setting it to `None` intentional?

Reply:
Currently (before this PR) if we called tight_layout it kills constrained_layout, permanently.  I think this is correct.  You don't want to call plt.tight_layout() but then have the previous engine run when the figure draws.  So I think this is right.  Rephrase the note to 

```
        # note that here we do not permanently set the figures engine to 
        # tight_layout but rather just perform the layout in place and remove 
        # any previous engines. 
```